### PR TITLE
Fetch swpc content with Python script

### PR DESCRIPTION
**Pull request by jeanconn**
Fetch swpc content with Python script

### CODE REVIEW

**Comment by jeanconn** [comment]
I tried changing web_content.cfg to have https urls for the swpc content, but still got 500 errors.
```
bash-4.1$ /data/fido/ska/bin/get_web_content.pl 
Warning: 500 SSL negotiation failed:  for web data solar_xray (https://services.swpc.noaa.gov/images/goes-xray-flux.gif)
Warning: 500 SSL negotiation failed:  for web data space_weather (https://services.swpc.noaa.gov/text/sgarf.txt)
```
My best guess is that the server is limiting to only newer versions of SSL and that our ska2 Perl Crypt::SSLeay etc is too old.  I figured it was not worth time to do detailed troubleshooting fix ska2, and that we can fix these files with Python and then decide later if we want to put them back as Perl fetches with skare3 Perl.

While I was at it, I set the new super-simple swpc script to fetch the ACE gif, as that is just a broken image on the mta source page at this time.

File: get_swpc.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+swpc_urls = {'sgarf.txt': 'https://services.swpc.noaa.gov/text/sgarf.txt',
+             'GOES_xray.gif': 'https://services.swpc.noaa.gov/images/goes-xray-flux.gif',
+             'ACE_5min.gif': 'http://services.swpc.noaa.gov/images/ace-epam-7-day.gif'}
+
+for locname, url in swpc_urls.items():
+    for _ in range(3):
+        try:
+            outfile = os.path.join(args.out, locname)
+            urllib.urlretrieve(url, outfile)
```
Comment (jeanconn) [inline get_swpc.py]: I don't remember which is the best Python 2 practice for a file fetch

File: web_content.cfg [hunk 2]
```diff
@@ -37,20 +37,23 @@
...
+    #    src = mta_ace_plot\.gif
+    #  </filter>
+    #</image>
   </ace>
 
   <space_weather>
-    url = http://services.swpc.noaa.gov/text/sgarf.txt
+    # sgarf.txt is fetched with get_swpc.py
+    #url = https://services.swpc.noaa.gov/text/sgarf.txt
+    url = file:///proj/sot/ska/ska/data/arc/sgarf.txt
```
Comment (jeanconn) [inline web_content.cfg]: This won't work in a dev ska but I couldn't remember what any of this config-templated file-parsing was doing so I just tried to hack in the independently updated text file.

File: web_content.cfg [hunk 3]
```diff
@@ -37,20 +37,23 @@
...
+    #    src = mta_ace_plot\.gif
+    #  </filter>
+    #</image>
   </ace>
 
   <space_weather>
-    url = http://services.swpc.noaa.gov/text/sgarf.txt
+    # sgarf.txt is fetched with get_swpc.py
+    #url = https://services.swpc.noaa.gov/text/sgarf.txt
+    url = file:///proj/sot/ska/data/arc/sgarf.txt
```
Comment (jeanconn) [inline web_content.cfg]: Fixed a typo here and lost the comment.  This won't work in a dev ska.

File: get_swpc.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            break
+        except Exception as err:
+            time.sleep(5)
+    else:
+        print 'Warning: failed to open URL {}: {}'.format(url, err)
+        sys.exit(0)
```
[the remark this answers is not in the document]
Comment (jeanconn) [reply get_swpc.py]: That seems a better idea, doesn't it.  Over-copied.

File: get_swpc.py (hunk 1 above)
Comment (jeanconn) [reply get_swpc.py]: OK.  I couldn't remember if they had complimentary functions (as in use requests to actually check status).  requests doesn't seem to implement a get-content-and-write-out function (unless I'm missing something in docs), and writing response.content seemed like it had to be a hack or mistake.

**Comment by jeanconn** [comment]
@taldcroft Should we make these changes?  I think mta has a plan to at least fix their fetch of the ACE plot, so we could use theirs or not as you like.

**Comment by jeanconn** [comment]
@taldcroft mta did fix their ACE plot (so I could remove it from this PR) but the GOES_xray plot isn't going to get better unless we do this fix or ask mta to start fetching it for us.

**Comment by jeanconn** [comment]
Hi @taldcroft, @swolk was asking about this.  Do we have a timetable?  Or should we remove the GOES plot from the list of checked files?  Or just keep muting gmail threads that are complaining about the broken file?

**Comment by taldcroft** [comment]
Have you put it into a dev ska and run?  Understood the one file won't work.

**Comment by jeanconn** [comment]
IIRC I tested the new script in a dev ska.  I started to make a new clean py2 dev ska to standalone test the full updated arc with the change, but then didn't see the point and stopped.

**Comment by taldcroft** [comment]
OK, approved.

**Comment by jeanconn** [comment]
Once I fix that comment, is this approved as @taldcroft  or FD?

**Comment by taldcroft** [comment]
Approved as FD.

**Comment by jeanconn** [comment]
Nah.  I basically live tested this and it doesn't work.  The new script picks up the files fine, but I didn't dig deep enough into how arc.pl finds those files to use them.  I'll revert this change and come up with something more comprehensive.

**Pull request merged**